Main/Search_Model.py

```diff
@@ -0,0 +1,43 @@
+import torch
+from Classes import GCN, BaseModel, EvaluationMetricsData
+from Train_Test import train, test
+from tqdm import trange
+from torch_geometric.loader import DataLoader
+
+
+# Basic search model that is used for testing parameters for each of the hyper parameter optimizers
+# Inputs:
+# params - library of parameters (str -> array)
+# train_dataset: train dataset containing graphs
+# test_dateset: test dataset containing graphs
+# device: contains device to run progam on
+
+def search_model(params, train_dataset, test_dataset, device) : 
+        # Define the data loaders, this uses batch_size
+        train_loader = DataLoader(dataset=train_dataset, batch_size=params["batch_size"], shuffle=True)
+        test_loader = DataLoader(dataset=test_dataset, batch_size=params["batch_size"], shuffle=False)
+
+        # Create model
+        gridModel = GCN(hidden_channels=params["hidden_channels"], dropout_rate=params["dropout_rate"], learning_rate=params["learning_rate"], 
+                        activation_function=params["activation_function"], amount_of_layers=params["amount_of_layers"], pooling_algorithm=params["pooling_algorithm"])
+        gridModel.to(device)
+
+        #Deciding which optimizer to use
+        optimizer = None
+        if(params["optimizer"].lower() == 'sgd'): optimizer = torch.optim.SGD(gridModel.parameters(), lr=gridModel.learning_rate)
+        elif(params["optimizer"].lower() == 'adam'): optimizer = torch.optim.Adam(gridModel.parameters(), lr=gridModel.learning_rate)
+        elif(params["optimizer"].lower() == 'rmsprop'): optimizer = torch.optim.RMSprop(gridModel.parameters(), lr=gridModel.learning_rate)
+        else : raise Exception("Invalid optimizer name: " + str(params["optimizer"]))
+
+        loss_function = torch.nn.CrossEntropyLoss()
+        baseModel = BaseModel(gridModel, loss_function, optimizer)
```

Comment:
^Should be its own function (create_model)

Reply:
you can do that sure